### PR TITLE
Proposal for the new website design and functionality

### PR DESCRIPTION
**Pull request by diamant3**
I want to propose this PR because:

- Everything is markdown format for editing
- Uses [github action](https://jekyllrb.com/docs/continuous-integration/github-actions/) to deploy the website
- Out of the box search feature

this is the demo, please access from here: https://diamant3.github.io/pspdev.github.io/
jekyll just-the-docs theme documentation: https://just-the-docs.com/

### CODE REVIEW

**Comment by sharkwouter** [comment]
I see this is not yet fully done, but I'll give you some feedback already. This looks really cool, thanks for working on this!

File: README.md
Comment (sharkwouter) [inline README.md:1]: It would be nice if the README actually contained the instructions on how to build locally.

File: basic_programs.md
Comment (sharkwouter) [inline basic_programs.md]: I would move this part to its own section on the left.

File: index.md
Comment (sharkwouter) [inline index.md:1]: This file should have a get started section with a link to the quick start page

File: quickstart.md
Comment (sharkwouter) [inline quickstart.md]: I would change it to something like this:
```suggestion
For more example take a look at the [Basic Programs](basic_programs.html) page.
```

File: basic_programs.md
Comment (sharkwouter) [inline basic_programs.md]:
```suggestion
title: Basic Programs
```

File: documentation.md
Comment (sharkwouter) [inline documentation.md]: I would rename this to Additional Documentation
```suggestion
title: Additional Documentation
```

File: tools/index.md
Comment (sharkwouter) [inline tools/index.md:1]: What would be added to tools? I would rather leave it out than leave it empty for now.

**Comment by sharkwouter** [comment]
Unrelated to this change, but it did make me think of this, we should make sure we keep the samples working with automated tests. I'll create an issue for it.

File: quickstart.md
Comment (sharkwouter) [inline quickstart.md]:
```suggestion
### Dependencies
```

File: basic_programs.md
Comment (sharkwouter) [inline basic_programs.md:9]: I would put the hello world which is also on the other page here as well.

**Comment by sharkwouter** [comment]
Also unrelated to this PR. Some of the examples could use slightly better descriptions. The SDL_mixer example does not describe that it's about audio for example.

**Comment by sharkwouter** [comment]
Can you mark this as draft until it's done?

File: downloads.md
Comment (sharkwouter) [inline downloads.md]: I would put this after quickstart

File: downloads.md
Comment (sharkwouter) [inline downloads.md]:
```suggestion
Downloading the latest version of the PSPDEV toolchain for your system [here](https://github.com/pspdev/pspdev/releases/tag/latest).

Alternatively, a Docker container is available [here](https://github.com/pspdev/pspdev/pkgs/container/pspdev).
```
This was using the wrong link for the docker container. It stating source code is also not quite right.

**Comment by diamant3** [comment]
Yes, for the demo. Sorry about that.😓 and thank you for your consideration wouter! This will be a huge change in our website. 

Sure I'll draft it for now, I will be working on it all this weekend.💪🏼

File: quickstart.md
Comment (sharkwouter) [inline quickstart.md]:
```suggestion
For more examples of programs take a look at the [Basic Programs](basic_programs.html) page.
```

File: basic_programs.md
Comment (sharkwouter) [inline basic_programs.md:1150]: I would add a link to the tips and tricks page at the end of this page.

File: quickstart.md
Comment (sharkwouter) [reply quickstart.md]: Sorry about making a typo here

**Comment by sharkwouter** [comment]
This is starting to look really good! Thanks for working on this!

**Comment by diamant3** [comment]
Thanks wouter! This would be more professional looking 😅

**Comment by diamant3** [comment]
In finalization, I will change the "use-jtd" to "master" in the push branches and this is the instruction https://jekyllrb.com/docs/continuous-integration/github-actions/#setting-up-the-action to become effective.

**Comment by diamant3** [comment]
about the tools section, I want to move the usage of psplink and other tools here in this website instead of per repo so we can focus each repo with the intro and description only, is it good, wouter?

**Comment by sharkwouter** [comment]
Adding the PSPLink information to the main website sounds like a good idea. I do think it is outside the scope of this PR, though.

**Comment by diamant3** [comment]
Got it wouter, thankss.

I'll mark it as ready for review now and do the necessary changes before merging

**Review by sharkwouter — APPROVED** [review]
This is looking good now. This is such a nice improvement, thanks for working on this!

I'm going to leave pressing the merge button to you. You can merge this whenever.

**Comment by diamant3** [comment]
Thanks again wouter for accepting!

**Pull request merged**
